security/pkg/caclient/clientrunner.go

```diff
@@ -0,0 +1,80 @@
+// Copyright 2018 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package caclient
+
+import (
+	"fmt"
+	"time"
+
+	"istio.io/istio/pkg/log"
+	pkiutil "istio.io/istio/security/pkg/pki/util"
+	"istio.io/istio/security/pkg/util"
+)
+
+// ClientRunner interacts with the upstream CA to maintain the KeyCertBundle valid.
+type ClientRunner interface {
+	// Run the ClientRunner loop (blocking call)
+	Run(stopCh <-chan struct{}, errCh chan<- error)
+}
+
+type clientRunner struct {
+	keycert  pkiutil.KeyCertBundle
```

Comment:
So we'll have one Runner instance per key/cert, and call Run() in a goroutine. If a NodeAgent has to refresh 200 service account's key, cert, does that mean to spawn 200 goroutines?

Can we make this libraries support multiple key/cert refresh, and in the loop, we find the earliest-to-be-expired key/cert 
 to send CSR?

Reply:
Thanks!